sacro/versioning.py

```diff
@@ -0,0 +1,68 @@
+import functools
+
+from django.conf import settings
+
+
+class IncorrectVersionError(Exception):
+    def __init__(self, *args, used, supported, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.used = used
+        self.supported = supported
+
+
+class UnsupportedVersionFormatError(Exception):
+    def __init__(self, *args, version, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.version = version
+
+
+@functools.total_ordering
+class Version:
+    """Utility class to parse and compare version strings"""
+
+    def __init__(self, version: str) -> None:
+        try:
+            major, minor, *_ = version.split(".")
+
+            # check major and minor are valid numbers
+            int(major)
+            int(minor)
+
+            self.major = major
+            self.minor = minor
+        except ValueError:
+            msg = f"Expected version to be in format 1.2.3, got {version}"
+            raise UnsupportedVersionFormatError(msg, version=version)
+
+        self.original = version
+
+    def __eq__(self, other: "Version") -> bool:
+        return self.major == other.major and self.minor == other.minor
+
+    def __gt__(self, other: "Version") -> bool:
+        if self.major > other.major:
+            return True
+
+        if self.major == other.major and self.minor > other.minor:
+            return True
+
+        return False
+
+    def __repr__(self):
+        return f"Version: {self.original}"
+
+    def __str__(self):
+        return self.original
+
+
+def check_version(version: str) -> None:
```

Comment:
This naming is potentially confusing, because we're specifically checking the version of the acro tool used to generate the config, whereas this could be that we're checking the version of "sacro" or something else.

I'm not sure what a better name might be.. `check_acro_version` perhaps, if it's not too verbose?

Reply:
I see your point but I think it's YAGNI, we only check one version in the app.  If we change that we can make the distinction!